boring/src/x509/mod.rs

```diff
@@ -229,16 +228,28 @@ impl X509Builder {
         }
     }
 
-    #[cfg(not(feature = "fips"))]
     /// Sets the notAfter constraint on the certificate.
     pub fn set_not_after(&mut self, not_after: &Asn1TimeRef) -> Result<(), ErrorStack> {
-        unsafe { cvt(X509_set1_notAfter(self.0.as_ptr(), not_after.as_ptr())).map(|_| ()) }
+        #[cfg(feature = "fips")]
+        unsafe {
+            cvt(X509_set_notAfter(self.0.as_ptr(), not_after.as_ptr())).map(|_| ())
+        }
+        #[cfg(not(feature = "fips"))]
+        unsafe {
+            cvt(X509_set1_notAfter(self.0.as_ptr(), not_after.as_ptr())).map(|_| ())
+        }
```

Comment:
Is `set_notAfter` still present in the latest version? Can we use that unconditionally instead of adding two different paths?

Reply:
I think so, yes - that way the same code gets tested whether the feature is enabled or disabled. Maybe add a comment that this should be changed when the version of FIPS is bumped (I'm not sure what the process for that is; I guess the C boring lib itself has to get recertified?).